RevenueCatUI/CustomerCenter/Views/AppUpdateWarningView.swift

```diff
@@ -0,0 +1,116 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  AppUpdateWarningView.swift
+//
+//  Created by JayShortway on 16/08/2024.
+
+#if CUSTOMER_CENTER_ENABLED
+
+import RevenueCat
+import SwiftUI
+
+#if os(iOS)
+
+@available(iOS 15.0, *)
+@available(macOS, unavailable)
+@available(tvOS, unavailable)
+@available(watchOS, unavailable)
+struct AppUpdateWarningView: View {
+    let onUpdateAppClick: () -> Void
+    let onContinueAnywayClick: () -> Void
+
+    @Environment(\.dismiss)
+    var dismiss
+
+    @Environment(\.localization)
+    private var localization: CustomerCenterConfigData.Localization
+    @Environment(\.appearance)
+    private var appearance: CustomerCenterConfigData.Appearance
+    @Environment(\.colorScheme)
+    private var colorScheme
+
+    @ViewBuilder
+    var content: some View {
+        ZStack {
+            if let background = Color.from(colorInformation: appearance.backgroundColor, for: colorScheme) {
+                background.edgesIgnoringSafeArea(.all)
+            }
+            let textColor = Color.from(colorInformation: appearance.textColor, for: colorScheme)
+
+            VStack {
+                CompatibilityContentUnavailableView(
+                    title: localization.commonLocalizedString(for: .updateWarningTitle),
+                    description: localization.commonLocalizedString(for: .updateWarningDescription),
+                    systemImage: "arrowshape.up.circle.fill"
+                )
+
+                Button(localization.commonLocalizedString(for: .updateWarningUpdate)) {
+                    onUpdateAppClick()
+                }
+                .buttonStyle(ProminentButtonStyle())
+                .padding(.bottom)
+
+                Button(localization.commonLocalizedString(for: .updateWarningIgnore)) {
+                    onContinueAnywayClick()
+                }
+            }
+            .padding(.horizontal)
+            .applyIf(textColor != nil, apply: { $0.foregroundColor(textColor) })
+        }
+        .toolbar {
+            ToolbarItem(placement: .navigationBarTrailing) {
```

Comment:
New compatibility view [incoming](https://github.com/RevenueCat/purchases-ios/pull/4188/files#diff-8ac72e331918e7a7c60cdf485ea66bbd4e8324a2320e99afd6be38552efd55d8), but it's not merged yet.

Reply:
Actually without this, I think `spm-revenuecat-ui-ios-15` will fail.

Reply:
Ah nice, using that cause it's merged now!